src/samplebuffer.h

```diff
@@ -0,0 +1,77 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h"
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+// The public SINTerface closely follows that of std::vector.
+//
+// No resize operation is provided SINTentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the desired
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer tempBuffer(newSize)
+//     ... copy data from sampleBuffer to tempBuffer...
+//     sampleBuffer.swap(sampleBuffer);
+//
+// The local variable tempBuffer can be omitted if no data needs to be
```

Comment:
.. because we should never call SampleBuffer tempBuffer(newSize) and swap within the engine code, since malloc can be blocking and is not a no op. So if one tries to resize the buffer it must be a very special issue. 
However, swap() might be nice, to re-factor gain = 1 copies.  We have to be careful that no one has a reference to m_data. 
I think that is worth a comment.